From d17441a4190d02ddd104baab3576507b74a406a9 Mon Sep 17 00:00:00 2001
From: Steve Loughran <stevel@apache.org>
Date: Thu, 23 Mar 2017 12:52:22 +0000
Subject: [PATCH 2412/2748] HADOOP-14120 needless
 S3AFileSystem.setOptionalPutRequestParameters in
 S3ABlockOutputStream putObject(). Contributed by
 Yuanbo Liu

(cherry picked from commit 20878d052cebc715c2494a97362fdff08885a77f)

Change-Id: If036da20e8f08f4f44c1c928a5ee423e84598169
---
 .../apache/hadoop/fs/s3a/S3ABlockOutputStream.java |    5 ++---
 .../org/apache/hadoop/fs/s3a/S3AFileSystem.java    |    2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java
index 1ac1f0d..f846689 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java
@@ -387,9 +387,8 @@ private int putObject() throws IOException {
     int size = block.dataSize();
     final S3ADataBlocks.BlockUploadData uploadData = block.startUpload();
     final PutObjectRequest putObjectRequest = uploadData.hasFile() ?
-        writeOperationHelper.newPutRequest(uploadData.getFile())
-        : writeOperationHelper.newPutRequest(uploadData.getUploadStream(), size);
-    fs.setOptionalPutRequestParameters(putObjectRequest);
+        writeOperationHelper.newPutRequest(uploadData.getFile()) :
+        writeOperationHelper.newPutRequest(uploadData.getUploadStream(), size);
     long transferQueueTime = now();
     BlockUploadProgress callback =
         new BlockUploadProgress(
diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
index 0cfaadd..50e7fe4 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
@@ -2070,7 +2070,7 @@ protected void setOptionalCopyObjectRequestParameters(
     }
   }
 
-  protected void setOptionalPutRequestParameters(PutObjectRequest request) {
+  private void setOptionalPutRequestParameters(PutObjectRequest request) {
     switch (serverSideEncryptionAlgorithm) {
     case SSE_KMS:
       request.setSSEAwsKeyManagementParams(generateSSEAwsKeyParams());
-- 
1.7.9.5

